core/src/main/java/io/questdb/std/ObjStack.java

```diff
@@ -24,93 +24,46 @@
 
 package io.questdb.std;
 
-import java.util.Arrays;
-
 public class ObjStack<T> implements Mutable {
-    private static final int MIN_INITIAL_CAPACITY = 8;
-    private static final int DEFAULT_INITIAL_CAPACITY = 16;
-    private T[] elements;
-    private int head;
-    private int tail;
-    private int mask;
+    private final ObjArrayDequeue<T> dequeue;
 
     public ObjStack() {
-        this(DEFAULT_INITIAL_CAPACITY);
+        dequeue = new ObjArrayDequeue<>();
     }
 
     public ObjStack(int initialCapacity) {
-        allocateElements(initialCapacity);
+        dequeue = new ObjArrayDequeue<>(initialCapacity);
     }
 
     public void clear() {
-        if (head != tail) {
-            head = tail = 0;
-            Arrays.fill(elements, null);
-        }
+        dequeue.clear();
     }
 
     public boolean notEmpty() {
-        return head != tail;
+        return dequeue.notEmpty();
     }
 
     public T peek() {
-        return elements[head];
+        return dequeue.peekLast();
     }
 
     public T peek(int n) {
-        return elements[(head + n) & mask];
+        return dequeue.peekLast(n);
```

Comment:
Same here, we need to have second delegate that doesn't seem to add any value.

Reply:
Reverted